design/nfs.md

```diff
@@ -0,0 +1,157 @@
+
+## Rook Architecture
+
+Rook turns distributed storage software into a self-managing, self-scaling, and self-healing storage services. It does this by automating deployment, bootstrapping, configuration, provisioning, scaling, upgrading, migration, disaster recovery, monitoring, and resource management. Rook uses the facilities provided by the underlying cloud-native container management, scheduling and orchestration platform to perform its duties.
+![Rook Architecture on Kubernetes](../Documentation/media/rook-architecture.png)
+
+## Network File System (NFS)
+
+NFS allows remote hosts to mount file systems over a network and interact with those file systems as though they are mounted locally. This enables system administrators to consolidate resources onto centralized servers on the network.
+
+## Why NFS?
+
+NFS is widely used for persistent storage in kubernetes cluster. Using NFS storage is a convenient and easy way to provision storage for applications.
+An NFS volume allows an existing NFS (Network File System) share to be mounted into the pod.
+The contents of an NFS volume are preserved and the volume is merely unmounted if the pod is stopped/destroyed. This means that an NFS volume can be pre-populated with data, and that data can be “handed off” between pods.
+NFS supports multiple read/write simultaneously so a single share can be attached to multiple pods.
+
+## Design
+
+Rook brings storage platforms to Kubernetes with a fully Kubernetes-native design. It integrates deeply into cloud native environments leveraging extension points and providing a seamless experience for scheduling, lifecycle management, resource management, security, monitoring, and user experience.
+With this project Rook is providing another widely adopted storage option for admins and users of cloud-native environments. This project will automate NFS starting from its configuration (such as allowed hosts, read/write permissions etc.) to deployment and provisioning. The operations on NFS which cannot be done natively by Kubernetes will be automated.
+NFS doesn’t provide an internal provisioner for kubernetes, so Rook is needed as an external provisioner.
+
+### Initial Setup
+
+The flow of creating NFS backed storage in Rook is
+1. The settings are determined and saved in an NFS server CRD (rook-nfs.yaml)
+2. `kubectl create -f rook-nfs.yaml`
+3. Rook is notified of the new CRD
+   - Rook starts the NFS daemon on the desired nodes in the cluster
+   - Rook creates a storage class
+4. The pod which requires NFS storage creates a PVC to consume the volume.
+
+### Deployed service
+
+When the NFS CRD instance is created, Rook responds to this request by starting the NFS daemon with the configuration and exports stated in the CRD and creates a service to expose NFS.
```

Comment:
Could this section be merged with step 3 in the previous section? There is some overlap.